src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -319,7 +320,13 @@ void runtime::gpu::GPU_ExternalFunction::emit_timer_functions()
                 }
             }
         }
-        m_writer << "ngraph::stopwatch timers[" << names.size() << "];\n";
+
+        if (m_shared_context->m_runtime_context->stopwatch_pool == nullptr)
+        {
+            m_shared_context->m_runtime_context->stopwatch_pool = new StopWatchPool;
+        }
+        m_offset = m_shared_context->m_runtime_context->stopwatch_pool->size();
+        m_shared_context->m_runtime_context->stopwatch_pool->allocate(names.size());
```

Comment:
nit:Any reason why the size of the stopwatch pool can't be sent in as an argument to the constructor instead of allocating the pool separately?

Reply:
since we put the pool in the shared context, functions can share the same pool, my thought is it will be more convenient to `allocate` instead of `construct`.

Reply:
Ah, ok got it, thanks.